src/tests/api-tests.ts

```diff
@@ -5393,6 +5394,171 @@ describe('api tests', () => {
+      block_height: 1,
+      source_code,
+      abi: '',
+    };
+    const dbBlock: DbBlock = {
+      block_hash: '0xd10ccecfd7ac9e5f8a10de0532fac028559b31a6ff494d82147f6297fb663139',
+      index_block_hash: '0xd10ccecfd7ac9e5f8a10de0532fac028559b31a6ff494d82147f6297fb663139',
+      parent_index_block_hash: '0xe7db9e4e93201756a94a14e7708ffa4ca45507b2d966791259acc8fcccb78565',
+      parent_block_hash: '0xe7db9e4e93201756a94a14e7708ffa4ca45507b2d966791259acc8fcccb78565',
+      parent_microblock_hash: '',
+      parent_microblock_sequence: -1,
+      block_height: 37979,
+      burn_block_time: 1637003433,
+      burn_block_hash: '0x0000000000000000000342c6f7e9313ffa6f0a92618edaf86351ca265aee1c7a',
+      burn_block_height: 709876,
+      miner_txid: '0x4321',
+      canonical: true,
+      execution_cost_read_count: 1210,
+      execution_cost_read_length: 1919542,
+      execution_cost_runtime: 2480886000,
+      execution_cost_write_count: 138,
+      execution_cost_write_length: 91116,
+    };
+    const expected = {
+      tx_id: '0x8407751d1a8d11ee986aca32a6459d9cd798283a12e048ebafcd4cc7dadb29af',
+      nonce: 0,
+      fee_rate: '139200',
+      sender_address: 'SPX3DV9X9CGA8P14B3CMP2X8DBW6ZDXEAXDNPTER',
+      sponsored: false,
+      post_condition_mode: 'allow',
+      post_conditions: [],
+      anchor_mode: 'any',
+      is_unanchored: false,
+      block_hash: '0xd10ccecfd7ac9e5f8a10de0532fac028559b31a6ff494d82147f6297fb663139',
+      parent_block_hash: '0xe7db9e4e93201756a94a14e7708ffa4ca45507b2d966791259acc8fcccb78565',
+      block_height: 37979,
+      burn_block_time: 1637003433,
+      burn_block_time_iso: '2021-11-15T19:10:33.000Z',
+      parent_burn_block_time: 1637002470,
+      parent_burn_block_time_iso: '2021-11-15T18:54:30.000Z',
+      canonical: true,
+      tx_index: 33,
+      tx_status: 'success',
+      tx_result: {
+        hex: '0x0100000000000000000000000000000001',
+        repr: 'u1',
+      },
+      microblock_hash: '',
+      microblock_sequence: 2147483647,
+      microblock_canonical: true,
+      event_count: 6,
+      events: [],
+      execution_cost_read_count: 55,
+      execution_cost_read_length: 88420,
+      execution_cost_runtime: 116256000,
+      execution_cost_write_count: 9,
+      execution_cost_write_length: 339,
+      tx_type: 'contract_call',
+      contract_call: {
+        contract_id: 'SP3YK7KWMYRCDMV5M4792T0T7DERQXHJJGGEPV1N8.pg-mdomains-v1',
+        function_name: 'bns-name-preorder',
+        function_signature:
+          '(define-public (bns-name-preorder (hashedSaltedFqn (buff 20)) (stxToBurn uint) (paymentSIP010Trait trait_reference) (reciprocityTokenTrait trait_reference) (referencerWallet principal)))',
+      },
+    };
+
+    await db.updateBlock(client, dbBlock);
+    await db.updateTx(client, tx1);
+    await db.updateTx(client, tx2);
+    await db.updateSmartContract(client, tx1, contractCall);
+    await db.updateSmartContract(client, tx2, contractCall2);
```

Comment:
Instead of making these updates directly, can you do this through the standard `update()` call? We're starting to use materialized views and other logic that needs to happen on update and I think it would be best if the code gets exercised as much as possible.